src/main/java/com/flightstats/hub/model/ChannelContentKey.java

```diff
@@ -25,10 +35,49 @@ public String toUrl() {
         return "channel/" + channel + "/" + contentKey.toUrl();
     }
 
-    public static ChannelContentKey fromUrl(String url) {
-        String after = StringUtils.substringAfter(url, "channel/");
-        String[] split = StringUtils.split(after, "/", 2);
-        return new ChannelContentKey(split[0], ContentKey.fromUrl(split[1]).get());
+    /**
+     * @param url Expects the format of ".+/channel/channelName/yyyy/mm/dd/hh/mm/ss/sss/hash"
+     */
+    public static ChannelContentKey fromResourcePath(String url) {
+        String channelPath = StringUtils.substringAfter(url, "channel/");
+        return fromChannelPath(channelPath);
+    }
+
+    /**
+     * @param path Expects the format of ".+/channelName/yyyy/mm/dd/hh/mm/[ss][sss][hash]"
+     */
+    public static ChannelContentKey fromSpokePath(String path) {
+        try {
+            String[] split = path.split("/");
```

Comment:
Hmm. The last element has to be handled specifically since it requires a second split (e.g. second, millisecond, and hash). I'll see if a hybrid of both approaches improves readability. Thanks for the tip.